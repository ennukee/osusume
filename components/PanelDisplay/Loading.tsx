import React from 'react';
import styled from 'styled-components';

interface LoadingProps {
  children: string;
  slow?: boolean;
}
export const Loading: React.FC<LoadingProps> = ({ children, slow }) => {
  return (
    <LoadingPanel>
      <LoadingLabel>{children}</LoadingLabel>
      <LoadingGif src="/static/loading.gif" />
      <SlowLabel>{slow && 'Stay tight, this part may take a few minutes to complete.'}</SlowLabel>
    </LoadingPanel>
  );
};

const LoadingPanel = styled.div`
    position: relative;
  min-height: inherit;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  padding: 20px;
`;

const LoadingLabel = styled.div`
    font-size: 16px;
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    font-weight: 600;
    text-align: center;
`;

const LoadingGif = styled.img`
    width: 150px;
`;

const SlowLabel = styled.div``;