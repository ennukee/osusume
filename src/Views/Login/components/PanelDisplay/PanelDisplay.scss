#login-state {
  display: flex;
  min-height: inherit;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  padding: 20px;

  a#login-button {
    background-color: var(--pink-light);
    font-size: 14px;
    width: 80%;
    text-align: center;
    border: 1px solid #222;
    border-radius: 3px;
    text-decoration: none;
    color: black;
    padding: 10px 20px;
    -webkit-border-radius: 3px;
    -moz-border-radius: 3px;
    -ms-border-radius: 3px;
    -o-border-radius: 3px;
  }
  #description {
    font-size: 14px;
    text-align: center;
  }
}
#header {
  font-size: 24px;
  // font-weight: bold;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}
