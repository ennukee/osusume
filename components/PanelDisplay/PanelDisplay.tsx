import React from 'react';
import { Loading } from './Loading';
import styled from 'styled-components';

interface PanelProps {
  phase: string;
}
export const PanelDisplay: React.FC<PanelProps> = ({ phase }) => {
  switch (phase) {
    case 'initialLoad':
      return <Loading>Checking for existing login...</Loading>;
    case 'login':
      return (
        <LoginStateContainer>
          <Header>Welcome</Header>
          <Description>
            Looks like you&apos;re new or have been logged out from your previous session. Use the button below to log
            back in!
          </Description>
          <LoginButton href="https://anilist.co/api/v2/oauth/authorize?client_id=1334&response_type=token">
            Log in
          </LoginButton>
        </LoginStateContainer>
      );
    case 'postLogin':
      return <Loading>Checking for osusume data...</Loading>;
    case 'generateProfile':
      return <Loading slow>Generating preference profile... (new user)</Loading>;
    case 'requestRecs':
      return <Loading slow>Picking out the best recommendations...</Loading>;
    case 'finishLogin':
      return <Loading>Done! Now redirecting...</Loading>;
    default:
      return <div>unknown</div>;
  }
};

const LoginButton = styled.a`
  background-color: var(--pink-light);
  font-size: 14px;
  width: 80%;
  text-align: center;
  border: 1px solid #222;
  border-radius: 3px;
  text-decoration: none;
  color: black;
  padding: 10px 20px;
  -webkit-border-radius: 3px;
  -moz-border-radius: 3px;
  -ms-border-radius: 3px;
  -o-border-radius: 3px;
`;

const Header = styled.div`
  font-size: 24px;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
`;

const Description = styled.div`
  font-size: 14px;
  text-align: center;
`;

const LoginStateContainer = styled.div`
  display: flex;
  min-height: inherit;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  padding: 20px;
`;
