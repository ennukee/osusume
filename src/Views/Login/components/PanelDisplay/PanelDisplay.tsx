import React from 'react';
import { Loading } from './components/Loading/Loading';
import './PanelDisplay.scss';

interface PanelProps {
  phase: string;
}
export const PanelDisplay: React.FC<PanelProps> = ({ phase }) => {
  switch (phase) {
    case 'initialLoad':
      return <Loading>Checking for existing login...</Loading>;
    case 'login':
      return (
        <div id="login-state">
          <div id="header">Welcome</div>
          <div id="description">
            Looks like you&apos;re new or have been logged out from your previous session. Use the button below to log
            back in!
          </div>
          <a href="https://anilist.co/api/v2/oauth/authorize?client_id=1334&response_type=token" id="login-button">
            Log in
          </a>
        </div>
      );
    case 'postLogin':
      return <Loading>Checking for osusume data...</Loading>;
    case 'generateProfile':
      return <Loading slow>Generating preference profile... (new user)</Loading>;
    case 'requestRecs':
      return <Loading slow>Picking out the best recommendations...</Loading>;
    case 'finishLogin':
      return <Loading>Done! Now redirecting...</Loading>;
    default:
      return <div>unknown</div>;
  }
};
