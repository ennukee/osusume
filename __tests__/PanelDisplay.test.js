import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { PanelDisplay } from '../components/PanelDisplay/PanelDisplay';

describe('Views/Login/components/PanelDisplay/PanelDisplay', () => {
  it('accepts initialLoad phase', () => {
    const { container } = render(<PanelDisplay phase={'initialLoad'} />);
    expect(container).toMatchSnapshot();
  });
  it('accepts login phase', () => {
    const { container } = render(<PanelDisplay phase={'login'} />);
    expect(container).toMatchSnapshot();

    // Link has correct attributes
    const link = container.querySelector('a');
    expect(link).toHaveTextContent('Log in');
    expect(link.getAttribute('href')).toBe(
      'https://anilist.co/api/v2/oauth/authorize?client_id=1334&response_type=token',
    );
  });
  it('accepts postLogin phase', () => {
    const { container } = render(<PanelDisplay phase={'postLogin'} />);
    expect(container).toMatchSnapshot();
  });
  it('accepts generateProfile phase', () => {
    const { container } = render(<PanelDisplay phase={'generateProfile'} />);
    expect(container).toMatchSnapshot();
  });
  it('accepts requestRecs phase', () => {
    const { container } = render(<PanelDisplay phase={'requestRecs'} />);
    expect(container).toMatchSnapshot();
  });
  it('accepts finishLogin phase', () => {
    const { container } = render(<PanelDisplay phase={'finishLogin'} />);
    expect(container).toMatchSnapshot();
  });
});
