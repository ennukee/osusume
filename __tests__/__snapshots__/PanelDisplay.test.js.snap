// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views/Login/components/PanelDisplay/PanelDisplay accepts finishLogin phase 1`] = `
<div>
  <div
    class="Loading__LoadingPanel-sc-1g2rmbs-0 bikQKX"
  >
    <div
      class="Loading__LoadingLabel-sc-1g2rmbs-1 gRBana"
    >
      Done! Now redirecting...
    </div>
    <img
      class="Loading__LoadingGif-sc-1g2rmbs-2 gbhczI"
      src="/static/loading.gif"
    />
    <div
      class="Loading__SlowLabel-sc-1g2rmbs-3 iAfWcn"
    />
  </div>
</div>
`;

exports[`Views/Login/components/PanelDisplay/PanelDisplay accepts generateProfile phase 1`] = `
<div>
  <div
    class="Loading__LoadingPanel-sc-1g2rmbs-0 bikQKX"
  >
    <div
      class="Loading__LoadingLabel-sc-1g2rmbs-1 gRBana"
    >
      Generating preference profile... (new user)
    </div>
    <img
      class="Loading__LoadingGif-sc-1g2rmbs-2 gbhczI"
      src="/static/loading.gif"
    />
    <div
      class="Loading__SlowLabel-sc-1g2rmbs-3 iAfWcn"
    >
      Stay tight, this part may take a few minutes to complete.
    </div>
  </div>
</div>
`;

exports[`Views/Login/components/PanelDisplay/PanelDisplay accepts initialLoad phase 1`] = `
<div>
  <div
    class="Loading__LoadingPanel-sc-1g2rmbs-0 bikQKX"
  >
    <div
      class="Loading__LoadingLabel-sc-1g2rmbs-1 gRBana"
    >
      Checking for existing login...
    </div>
    <img
      class="Loading__LoadingGif-sc-1g2rmbs-2 gbhczI"
      src="/static/loading.gif"
    />
    <div
      class="Loading__SlowLabel-sc-1g2rmbs-3 iAfWcn"
    />
  </div>
</div>
`;

exports[`Views/Login/components/PanelDisplay/PanelDisplay accepts login phase 1`] = `
<div>
  <div
    class="PanelDisplay__LoginStateContainer-s33gcz-3 eknvRK"
  >
    <div
      class="PanelDisplay__Header-s33gcz-1 dmvHNS"
    >
      Welcome
    </div>
    <div
      class="PanelDisplay__Description-s33gcz-2 dhjHFp"
    >
      Looks like you're new or have been logged out from your previous session. Use the button below to log back in!
    </div>
    <a
      class="PanelDisplay__LoginButton-s33gcz-0 dckaMb"
      href="https://anilist.co/api/v2/oauth/authorize?client_id=1334&response_type=token"
    >
      Log in
    </a>
  </div>
</div>
`;

exports[`Views/Login/components/PanelDisplay/PanelDisplay accepts postLogin phase 1`] = `
<div>
  <div
    class="Loading__LoadingPanel-sc-1g2rmbs-0 bikQKX"
  >
    <div
      class="Loading__LoadingLabel-sc-1g2rmbs-1 gRBana"
    >
      Checking for osusume data...
    </div>
    <img
      class="Loading__LoadingGif-sc-1g2rmbs-2 gbhczI"
      src="/static/loading.gif"
    />
    <div
      class="Loading__SlowLabel-sc-1g2rmbs-3 iAfWcn"
    />
  </div>
</div>
`;

exports[`Views/Login/components/PanelDisplay/PanelDisplay accepts requestRecs phase 1`] = `
<div>
  <div
    class="Loading__LoadingPanel-sc-1g2rmbs-0 bikQKX"
  >
    <div
      class="Loading__LoadingLabel-sc-1g2rmbs-1 gRBana"
    >
      Picking out the best recommendations...
    </div>
    <img
      class="Loading__LoadingGif-sc-1g2rmbs-2 gbhczI"
      src="/static/loading.gif"
    />
    <div
      class="Loading__SlowLabel-sc-1g2rmbs-3 iAfWcn"
    >
      Stay tight, this part may take a few minutes to complete.
    </div>
  </div>
</div>
`;
